index.js

```diff
@@ -159,4 +159,49 @@ mung.headersAsync = function headersAsync (fn) {
     }
 }
 
+mung.write = function write (fn, options = {}) {
+    return function (req, res, next) {
+        const original = res.write;
+        const mungError = options.mungError;
+
+        res.write = function (chunk, encoding, ...args) {
+            if (res.headersSent) {
+                return;
+            }
+
+            // Do not mung on errors
+            if (!mungError && res.statusCode >= 400) {
+                return original.apply(res, arguments);
+            }
+
+            try {
+                let modifiedChunk = fn(chunk, req, res);
```

Comment:
Let's pass the encoding to the mung function.  Make sure `null` is passed if its really the callback.

Something like
```
let enc = typeof encoding === 'function' ? null : enc;
let modifiedChunk = fn(chunk, enc, req, res);